zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -1608,11 +1656,19 @@ object ZQuery {
     succeed(Some(a))
 
   /**
-   * Constructs a query that succeeds with the specified value.
+   * Lazily constructs a query that succeeds with the specified value.
+   *
+   * '''NOTE''': If the `value` is side-effect free, prefer using [[succeedNow]] instead
    */
   def succeed[A](value: => A)(implicit trace: Trace): ZQuery[Any, Nothing, A] =
     ZQuery(ZIO.succeed(Result.done(value)))
 
+  /**
+   * Eagerly constructs a query that succeeds with the specified value.
+   */
+  def succeedNow[A](value: A): ZQuery[Any, Nothing, A] =
```

Comment:
Are we ok with having this part of the public API? I know historically the core has made `*Now` stuff package private to avoid footguns.

Reply:
Personally I think it's okay. Most (if not all) other frameworks, including Scala's Future (which I know isn't the best example because it's not referentially transparent) expose methods for eagerly constructing monads. I think in ZIO having the `*Now` methods makes less sense because it has `Exit`. In ZQuery users don't really have that option.